src/NServiceBus.RabbitMQ/Routing/DefaultRoutingKeyConvention.cs

```diff
@@ -21,6 +22,12 @@ static string GetRoutingKey(Type type, string key = "")
             var interfaces = type.GetInterfaces()
                 .Where(i => !IsSystemType(i) && !IsNServiceBusMarkerInterface(i)).ToList();
 
+            if (interfaces.Count > 1)
+            {
+                Logger.WarnFormat(@"The direct routing topology may not deliver message types with interfaces in their inheritance hierarchy. Refrain from using interface
```

Comment:
Given that we're only logging this when we find more than one interface, it feels like maybe the message should be more specific here, pointing out that we're logging it because there is more than one?

On the other hand, if we want to prevent people from using any interfaces at all, maybe the `if` should be changed to > 0? It feel like the current error message is assuming that it would be logged if any are found.